packages/server/src/server.ts

```diff
@@ -146,22 +146,26 @@ const logAndHandleResponse = async (methodName, methodFunction) => {
     ms = Date.now() - start;
     methodResponseHistogram.labels(methodName, status).observe(ms);
     logger.info(`${messagePrefix} ${status} ${ms} ms `);
+    if (response instanceof JsonRpcError) {
+      return new JsonRpcError({name: response.name, code: response.code, message:`${requestIdPrefix} ` + response.message});
+    } 
     return response;
   } catch (e: any) {
     ms = Date.now() - start;
     methodResponseHistogram.labels(methodName, responseInternalErrorCode).observe(ms);
     logger.error(e, `${messagePrefix} ${responseInternalErrorCode} ${ms} ms`);
 
+    let error = predefined.INTERNAL_ERROR;
     if (e instanceof MirrorNodeClientError) {
       if (e.isTimeout()) {
-        return predefined.REQUEST_TIMEOUT;
+        error = predefined.REQUEST_TIMEOUT;
       }
     }
     else if (e instanceof JsonRpcError) {
-      return e;
+      error = e;
     }
 
-    return predefined.INTERNAL_ERROR;
+    return new JsonRpcError({name: error.name, code: error.code, message:`${requestIdPrefix} ` + error.message});
```

Comment:
Q: Would it have been easier to just add a param value to `predefined.INTERNAL_ERROR` that takes a requestId?